dataset/src/test/scala/frameless/TypedDatasetSuite.scala

```diff
@@ -6,11 +6,7 @@ import org.scalactic.anyvals.PosZInt
 import org.scalatest.FunSuite
 import org.scalatest.prop.Checkers
 
```

Comment:
I agree. `Job[_]` is a type generator with a single hole, that shouts to be seen as a monad, but honestly it's hard for me to explain to others why we need to wrap the lazy TypedDataset into yet another layer. At least we should give an appealing reason why someone needs to go from: 

```scala
// vallila Spark
ds.count()

// frameless
ds.count().run()
```

Maybe not the best place to open the discussion, but since I saw Job on the PR title, why not :D`